src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -57,61 +50,74 @@ public static boolean isValidDouble(Object value){
+
         }
-        return categoryOB.success_add(categoryOD);
+        return categoryOB.success_add();
+
     }
     /**
      * Overrides method in Category_IB.
-     * Attempts to edit a category with information from CategoryID and returns whether fail/success after execution.
-     * Provides detailed fail messages according to each condition listed below
-     *  1. NumberFormatException: User tries to edit a budget value with input that can not be converted to a double.
-     *  2. Category budget less than 0: User tries to edit a budget value with input that is a negative number.
-     *  3. EntityException: User tries to edit a Category with invalid input but failed. (See entities/EntityException.java)
+     * Attempts to edit a category with information from CategoryID and returns a CategoryOD indicating whether fail/success after execution.
+     * Provides detailed fail messages according to each condition listed below:
+     *  1. Repeated Name: User tries to edit category name to another name that exists in the month.
+     *  2. NoSuchElementException: User tries to edit a category that does not exist.
+     *  3. NumberFormatException: User tries to edit a budget value with input that can not be converted to a double.
+     *  4. Category budget less than 0: User tries to edit a budget value with input that is a negative number.
      *
      * @param categoryID_edit Category_Input_Data required for editing a new category to the designated monthID MonthlyStorage Object.
      * @return Category_Output_Data category.
-     * @throws EntityException thrown when categoryID_add has invalid category information.
      */
     @Override
     public CategoryOD editCategoryInMonth(CategoryID categoryID_edit) throws EntityException {
         MonthlyStorage month = categoryID_edit.getSession().getMonthlyData(categoryID_edit.getMonthID());
         ArrayList<Category> category_list = month.getCategoryData();
         try {
-            Category c = findCategory(category_list, categoryID_edit.getOld_category().getName());
-
-            c.setName(categoryID_edit.getName());
-            if(isValidDouble(categoryID_edit.getValue())){
-                double value_double = Double.parseDouble(String.valueOf(categoryID_edit.getValue()));
-                categoryID_edit.setValue(value_double);
+            for (Category category1 : category_list) {
+                if (category1.getName().equals(categoryID_edit.getName())) {
+                    //1. Repeated name fail
+                    return categoryOB.fail("There is already a category with this new name in this month.");
+                }
             }
-            if ((double)categoryID_edit.getValue() < 0) {
-                categoryOB.fail("Category budget can't be less than $0. Please try again!");
+
+            Category existing_category = findCategory(category_list, categoryID_edit.getOld_category());
+
+            double value_double = toDouble(categoryID_edit.getValue());
```

Comment:
These are also in pothole case. You will likely need to control f '_' on your project to find them all.